src/bindfs.c

```diff
@@ -1478,11 +1655,24 @@ static int bindfs_create(const char *path, mode_t mode, struct fuse_file_info *f
     int fd;
     struct fuse_context *fc;
     char *real_path;
+    int access_check_result;
+    int path_search_permission;
 
     real_path = process_path(path, true);
     if (real_path == NULL)
         return -errno;
 
+    access_check_result = dirname_access_check(real_path, W_OK);
+    if (access_check_result != 0)
+        return access_check_result;
+
+    path_search_permission = check_path_search_perms(real_path);
+    if (path_search_permission != 0)
+        return path_search_permission;
+
+    // TODO: Pretty sure we need more access checks here. Will let the tests
+    // highlight the issues.
```

Comment:
Note to self: now that I look at this function, it's almost identical to `bindfs_open`. Should probably factor out a helper.